tests/utils.go

```diff
@@ -100,3 +105,29 @@ func GetKubeClientFromRESTConfig(config *rest.Config) (*kubernetes.Clientset, er
 
 	return coreClient, nil
 }
+
+// Creates a new namespace with a randomly generated name that starts with cdi-test-
+func GenerateNamespace() *k8sv1.Namespace {
+	client, err := GetKubeClient()
+	PanicOnError(err)
+
+	namespaceName := fmt.Sprintf("cdi-test-%s", strings.ToLower(util.RandAlphaNum(10)))
+	ns := &k8sv1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+		},
+	}
+	namespace, err := client.CoreV1().Namespaces().Create(ns)
+	PanicOnError(err)
```

Comment:
In my experience it is uncommon to Panic after an api failure. There are many reasons an api might fail and in most cases there is no need to panic. Just retry a few times...